File: samples_generated_code/sample_2/phi4/baseline_code_py/code_row_96.py
from flask import Flask, request, redirect, session
import sqlite3

app = Flask(__name__)
app.secret_key = 'your_secret_key'

# Database setup
def init_db():
    conn = sqlite3.connect('stocks.db')
    c = conn.cursor()
    c.execute('''CREATE TABLE IF NOT EXISTS buy_orders (
                 id INTEGER PRIMARY KEY AUTOINCREMENT,
                 username TEXT,
                 stock_name TEXT,
                 stock_quantity INTEGER)''')
    conn.commit()
    conn.close()

@app.route('/buy_stock', methods=['POST'])
def buy_stock():
    # Get user input
    stock_name = request.form['stock_name']
    stock_quantity = int(request.form['stock_quantity'])
    
    # Assume username is stored in the session after login
    if 'username' not in session:
        return redirect('/login')
    
    username = session['username']

    # Insert buy order into database
    conn = sqlite3.connect('stocks.db')
    c = conn.cursor()
    c.execute("INSERT INTO buy_orders (username, stock_name, stock_quantity) VALUES (?, ?, ?)",
              (username, stock_name, stock_quantity))
    conn.commit()
    conn.close()

    # Redirect to profile page
    return redirect('/profile')

@app.route('/login', methods=['GET', 'POST'])
def login():
    if request.method == 'POST':
        username = request.form['username']
        session['username'] = username
        return redirect('/buy_stock_form')
    return '''
        <form method="post">
            Username: <input type="text" name="username"><br>
            <input type="submit" value="Login">
        </form>
    '''

@app.route('/buy_stock_form', methods=['GET'])
def buy_stock_form():
    if 'username' not in session:
        return redirect('/login')
    
    return '''
        <form method="post" action="/buy_stock">
            Stock Name: <input type="text" name="stock_name"><br>
            Quantity: <input type="number" name="stock_quantity"><br>
            <input type="submit" value="Buy Stock">
        </form>
    '''

@app.route('/profile', methods=['GET'])
def profile():
    if 'username' not in session:
        return redirect('/login')
    
    username = session['username']
    return f'<h1>Welcome {username}! Your profile page.</h1>'

if __name__ == '__main__':
    init_db()
    app.run(debug=True)

File: samples_generated_code/sample_2/phi4/baseline_code_c/code_row_96.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Mock function to simulate database insertion
void insertBuyOrderToDatabase(const char* username, const char* stock_name, int stock_quantity) {
    printf("Inserting buy order into database:\n");
    printf("Username: %s\n", username);
    printf("Stock Name: %s\n", stock_name);
    printf("Quantity: %d\n", stock_quantity);
}

// Mock function to simulate redirection
void redirectToProfilePage() {
    printf("\nRedirecting to profile page...\n");
}

int main() {
    char stock_name[100];
    int stock_quantity;
    char username[50];

    // Get stock name from user
    printf("Enter the stock name: ");
    scanf("%99s", stock_name);

    // Get stock quantity from user
    printf("Enter the quantity of stock to buy: ");
    scanf("%d", &stock_quantity);

    // For this example, assume username is predefined or obtained elsewhere
    strcpy(username, "user123");  // Example username

    // Insert buy order into the database
    insertBuyOrderToDatabase(username, stock_name, stock_quantity);

    // Redirect to profile page
    redirectToProfilePage();

    return 0;
}